cachi2/core/errors.py

```diff
@@ -96,33 +96,22 @@ class FetchError(Cachi2Error):
     )
 
 
-class GoModError(Cachi2Error):
-    """The 'go' command used while processing a Go module returned an error.
+class SubProcessError(Cachi2Error):
```

Comment:
The name suggests it's a subclass of a subprocess error which it's not. Can we make it even more generic and call it perhaps `PackageManagerError`. We could then subclass this one further if needed for a specific package manager. The good thing about that would be that we could then choose during exception handling if we want to catch a specific error, or any package manager error or any Cachi2Error.

Reply:
Yeah, I made it SubProcess error in case we wanted to wrap other subprocesses in it (like Git or whatever else we end up calling). But I like the idea of `PackageManagerError`.

Reply:
I like PackageManagerError as well :+1: